lib/autocomplete.js

```diff
@@ -0,0 +1,90 @@
+const _ = require("lodash");
+const aws = require("aws-sdk");
+const parsers = require("./parsers");
+const core = require("./core");
+const consts = require("./consts.json");
+
+/**
+ * Maps parameters passed to Autocomplete function to an object with parsed values.
+ *
+ * @param params: [ { value: any, name: string, type: string } ]
+ * Array of values passed to autocomplete function as action parameters.
+ *
+ * @returns {*} - object with params names as keys and parsed values
+ */
+function parseActionParameters(params) {
+  return params.reduce((acc, {
+    value, name, type, valueType,
+  }) => ({
+    ...acc,
+    [name]: parsers.resolveParser(type || valueType)(value),
+  }), {});
+}
+
+function handleInput(awsService, actionParams, pluginSettings) {
+  const [params, settings] = [actionParams, pluginSettings].map(parseActionParameters);
+  const client = core.getServiceInstance(awsService, params, settings);
+  return { params, settings, client };
+}
+
+function simpleItemsFilter(autocompleteItems, query) {
+  let result = autocompleteItems;
```

Comment:
Result gets overwritten anyway, so this initialization doesn't save us anything. It's more readable when the processing is performed on the `autocompleteItems` and then its result saved to `result`.

Reply:
I kind of agree, but also this whole function is just bad for so many different reasons. This looks like this because the `query` was not being passed to the autocomplete functions for the longest time as a result of the bug, so this function was pretty much taken 1:1 from the original code with only a minor adjustments and was intended to be refined after the bug with query is fixed. AFAIK it's fixed now and I will work on this code. Sorry, I should have marked it somehow with @fixme or whatever.

As for your comment – it's almost true, with one little problem: `sort` is destructive in js, meaning it sorts the array in-place instead of creating the copy. I reassign the result for the sorted version for clarity, it's not necessary here, but if I didn't reassign `autocompleteItems` to `result` it would change the original parameter which is a bad way to do stuff in general.  But as I mentioned, I will replace this whole function.